oracle_fdw.c

```diff
@@ -890,6 +950,23 @@ oracleGetForeignPaths(PlannerInfo *root, RelOptInfo *baserel, Oid foreigntableid
 	if (usable_pathkeys != NIL)
 		fdwState->order_clause = orderedquery.data;
 
+	/*
+	 * Add LIMIT (FETCH N FIRST ROW ONLY) expression if not OFFSET clause
+	 * is used and that oracle version >= 12.
+	 */
```

Comment:
I think it would be fine to push down `LIMIT` even in the face of an `OFFSET`.
Then the limit passed to Oracle should be the sum of both values.
That would still give the Oracle optimizer the right clue.

Reply:
yeah, good idea.